docs/docs/gatsby-magic.md

```diff
@@ -0,0 +1,88 @@
+
+Plugins are different from starters in that plugins package up functionality for Gatsby’s data layer and other structural site code into installable parts. Starters, in contrast, include all source code for a Gatsby site including CSS styles and JSX markup.
+
+To learn more about Gatsby’s plugin system, check out the [plugins section](https://www.gatsbyjs.org/docs/plugins/) of the docs.
+
+##
+
+Themes
+
+Gatsby themes are a type of plugin with a `gatsby-config.js` file, effectively making them into Gatsby sites you can install from a package manager like npm. Themes provide a way to package up and distribute functionality for both Gatsby’s data layer and UI components, making them possible to update and receive changes from the original source code. This is how themes differ from starters, which do not maintain a connection to their source. Themes also come with a powerful API called [Shadowing](https://www.gatsbyjs.org/docs/themes/shadowing/), which allows you to structure themes in a customizable hierarchy.
+
+Themes for various use cases are added to the Gatsby ecosystem daily, such as documentation themes, UI library implementations, and more. It’s also possible to create your own theme and either distribute it for people in the community to use or keep it for internal use.
+
+To learn more about Gatsby themes including themes available in the plugin library, check out the [themes section](https://www.gatsbyjs.org/docs/themes/) of the docs.
+
+## Routing
+
+Gatsby sites are full React web applications, meaning they have access to client-side APIs including routing (a fancy word for matching a URL with a part of a web page using JavaScript). Gatsby creates paths to access content, handling dynamic [routing](https://www.gatsbyjs.org/docs/glossary#routing) for you and providing performance benefits through [content prefetching](https://www.gatsbyjs.org/docs/routing/#performance-and-prefetching). A lot of the work is done at build time for a site using mostly [static content](https://www.gatsbyjs.org/docs/adding-app-and-website-functionality/#static-pages). To handle applications that include functionality that cannot be handled at build time or through [rehydration](https://www.gatsbyjs.org/docs/adding-app-and-website-functionality/#how-hydration-makes-apps-possible), including authentication or retrieving dynamic content, you can use [client-only routes](https://www.gatsbyjs.org/docs/client-only-routes-and-user-authentication) using [@reach/router](https://www.gatsbyjs.org/docs/reach-router-and-gatsby/) which is built into Gatsby.
+
+Gatsby makes it easy to create HTML pages that hook into its routing system. Here are some examples of [how to create pages](https://www.gatsbyjs.org/docs/creating-and-modifying-pages) in Gatsby:
```

Comment:
```suggestion
Gatsby provides a few different ways to create HTML pages that hook into its routing system. Here are some examples of [how to create pages](https://www.gatsbyjs.org/docs/creating-and-modifying-pages) in Gatsby:
```
Even though it's Gatsby making it easy I still felt like we should remove the "easy" part. Just in case someone finds it challenging!